README.md

```diff
@@ -0,0 +1,131 @@
+
+## Local Development
+This is project is built with Laravel & Vue.js. Before you can start local development you need to take care of a few prerequisites, required by laravel.
+
+### Installing Prerequisites
+You can find the server prequisites listed in [laravel docs](https://laravel.com/docs), Additionally, you would require to install [composer](https://getcomposer.org/) & [nodejs](https://nodejs.org/en/) to pull in all the project dependencies. For Linux, you can follow the instructions below to get all the prerequisites under the hood. 
+> ##### Note:
+> Please note that these instructions are not tested yet, and *may not* work in first place.
+
+##### Using `apt` package manager (Debian/Ubuntu)
+Before you begin installing make sure you run `sudo apt update` to get the latest version available.
+
+```bash
+
+# if you do not have mysql installed on your system
+sudo apt install mysql-server
+
+# php & required extensions
+sudo apt install php7.2 php7.2-mysql php7.2-xml php7.2-mbstring php7.2-bcmath php7.2-sqlite php7.2-json
+
+# composer & nodejs
+sudo apt install nodejs composer
+```
+
+##### Using `pacman` package manager (ArchLinux, Manjaro, Antergos)
+Before you begin installing make sure you run `sudo pacman -Sy` to get the latest version available.
+
+```bash
+# if you do not have mysql installed on your system
+sudo pacman -S mysql
+
+# php & required extensions
+sudo pacman -S php php-mysql php-xml php-mbstring php-bcmath php-sqlite php-json
+
+# composer & nodejs
+sudo pacman -S nodejs composer
+```
+
+##### Make sure you have pdo extension enabled in `php.ini`
+
+Locate your php ini file using `php --ini`. For Ubuntu, it would be something like `/etc/php/7.2/cli/php.ini` and for Archlinux it is something like `/etc/php/php.ini`.
+
+Search for the text `pdo-mysql` in this file if this line is commented i.e it is preceded with `;`, make sure you uncomment it by removing `;`.
+
+
+### Clone Project
+You can simply clone the project using `git` as:
+
+```bash
+git clone https://github.com/csdu/chakravyuh
+```
+
+or you can simply download ZIP and extract it.
+
+### Install project dependencies
+
+Change into project directory and install php dependencies using `composer`:
+
+```bash
+composer install
+```
+
+This will install all the php packages required by the project. Similarly, we would use `npm` to install all the required JavaScript dependencies.
+
+```bash
+npm install
+```
+
+This will install all the JavaScript packages required by the project. After installing JavaScript dependencies, you should compile down the frontend assets using,
+
+```
+npm run dev
+```
+
+This will compile down our stylesheets (CSS) & javascript files. You can also run a watcher to automatically compile the assets, whenever the files are changed. This is recommended when you're working with SASS or the javascript, you do not need to run `npm run dev` again and again, you can simply run:
+
+```
+npm run watch
+```
+
+### Application Configuration
```

Comment:
Also mention the setup for facebook and google Authentication (AppId, AppSecret).